cpp/src/utilities/validation.cuh

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <iostream>
+
+#include <cuda.h>
+#include <cuda_runtime_api.h>
+
+#include <cudf/types.h>
+#include "nvgraph_error_utils.h"
+#include <thrust/sort.h>
+
+namespace cugraph {
+// Function for checking 0-based indexing
+template <typename T>
+gdf_error indexing_check (T* srcs, T* dests, int64_t nnz) {
+    cudaStream_t stream {nullptr};
+
+    // min from srcs 
+    auto minId_it = thrust::min_element(rmm::exec_policy(stream)->on(stream), srcs, srcs + nnz);
+    T minId;
+    CUDA_TRY(cudaMemcpy(&minId, minId_it, sizeof(T), cudaMemcpyDefault));
+    // negative index are not allowed
+    if (minId < 0 )
+        return GDF_INVALID_API_CALL; 
+    
+    minId_it = thrust::min_element(rmm::exec_policy(stream)->on(stream), dests, dests + nnz);
+    T minId2;
+    CUDA_TRY(cudaMemcpy(&minId2, minId_it, sizeof(T), cudaMemcpyDefault));        
+    // negative index are not allowed
+    if (minId2 < 0 )
+        return GDF_INVALID_API_CALL; 
+
+    minId = minId < minId2 ? minId : minId2;
+
+    // warning when smallest vertex is not 0
+    if (minId > 0 ) {
+        std::cerr<< "WARNING: the smallest vertex index in the edge set is "<<minId<<". ";
+        std::cerr<< "Cugraph supports 0-based indexing. ";
+        std::cerr<< "Vertex 0 to "<<  minId <<" will be created." << std::endl;
```

Comment:
I cannot understand what "Vertex 0 to minId will be created" actually means. Can this warning be little more specific?

Reply:
Perhaps another place where [0, minId) would be appropriate.  Or is this [0, minId]?

Reply:
@seunghwak here are more details : 
The smallest vertex is always 0 in cugraph. When 0 does not have any neighbor (ie. 0 cannot be found in src or dst) we will show a warning saying that vertex [0, <smallest ID found in the edge list>) still exist. In most cases, that's expected. However, some user may be using a 1-based indexing or random indexes starting with several digits. This warning will help them identify that they should use the renumbering feature prior to loading the edge list.

I updated the comment to make this clearer while trying to keeping it concise.